### ISSUE TITLE
Virtual networks should've been using 'route' mode instead of 'nat'

### ISSUE THREAD

**Issue by bashtheshell** [bashtheshell/jang_homelab#13]
In the past, I have had issues with the virtual networks, using NAT, not being able to communicate with each other until I used a libvirt hook script to fix this problem. However, I didn't foresee this being much of an issue until I started working with the server configurations as part of the RHCE objectives.

While the four VMs were able to communicate with each other, the VMs from one network would have to use theirs gateway to reach the VM in another network. This is normal by design. However, here is a situation. Let's say `tester1` in 192.168.122.0/24 network wants to reach `outsider1` in 192.168.100.0/24 network. `tester1` would sent the packet to its gateway as it should have a route to forward the `tester1`'s packet to `outsider1`. The `tester1`'s gateway forwards the packet to `outsider1`'s gateway as its claimed to have a route for `outsider1` in its local network.

While in practice, this is great. This is how packets should traverse through the networks in general. However, because we are dealing with two NAT networks, the packets' header would be modified once it reaches the destination's gateway. So when the destination host (in our case which is `outsider1`) receives the packets, it'd appears as if it's coming from its gateway rather than the source. I can assure you this is not a bug as it's normal by design. It's just that the NAT implementation for the purpose of this lab isn't appropriate.

As you can imagine, I'd have to modify the services' ACL rules to allow only the gateway to pass the appropriate requests to it as well as the firewall rules for the host. This is a security issue as we have already explicitly allowed all traffics through the VM networks with the hook scripts. So I have looked into an option to retain most of the networking behavior other than the NAT add-on, which I don't need. The 'route' mode came close to what I anticipated as it'd not strip the IP headers. However, the VMs won't be able to access the Internet directly without any modification to the hypervisor host's IP stack or the LAN's router/gateway that the hypervisor is directly connected to. This would also mean that this lab would now be restricted to using local repositories configured on the hypervisor host with its out-of-the-box configuration. 

I would need to make a small quick update to the [libvirt-prep-fb.sh](https://github.com/bashtheshell/jang_homelab/blob/534a4037e419391b2fe0d1cbf15689b7bac5fb75/libvirt-prep-fb.sh) script to change the mode to 'route'. I also plan to comment out the QEMU code blocks on the bottom of the (build-vms-fb.sh)[https://github.com/bashtheshell/jang_homelab/blob/534a4037e419391b2fe0d1cbf15689b7bac5fb75/build-vms-fb.sh] script.

This is quite a learning experience, and I can't say I've learned my lessons as there are more unforeseeable consequences that come with this change. Users preparing for the RHCSA may not appreciate this change with the lack of Internet connectivity out of the box, and I apologize in advance for the inconvenience. I hope to come back here eventually to find a solution to fix this.

**Comment by bashtheshell** [comment]
On a second thought, I already ran into a problem. I realize the guest VMs really should have access to the Internet. The NTP and DNS servers would need external access. Although, I can also restrict their access to only query the local servers installed on the hypervisor.

**Comment by bashtheshell** [comment]
Ultimately, I've decided to configure the NTP and DNS servers on the hypervisor to act as a local proxy for the VMs. The fixes have been made. This is still not what I wanted, but due to the time constraint, I'm taking this approach instead.

I've entertained the idea of moving the VM bridges (VM networks) to another new bridge and set up a NAT for that bridge network. Ideally, the main bridge would have the routes for all sub-bridges (VM networks). This is just an idea as I still don't have a clue what could be the implications of this network setup. Would the VMs still communicate with the hypervisor host transparently or do I need to perform port forwarding? If nothing goes well eventually, then the users would have to use an appropriate router for their LAN network with the static routes capability. That's not a burden I'd want to put on anyone.
